pulp_rpm/app/tasks/prune.py

```diff
@@ -0,0 +1,156 @@
+from datetime import datetime, timedelta
+from logging import getLogger, DEBUG
+
+from django.db.models import F, Subquery
+from django.utils import timezone
+
+from pulpcore.plugin.models import ProgressReport
+from pulpcore.plugin.constants import TASK_STATES
+from pulpcore.plugin.models import (
+    GroupProgressReport,
+    RepositoryContent,
+    TaskGroup,
+)
+from pulpcore.plugin.tasking import dispatch
+from pulp_rpm.app.models.package import Package
+from pulp_rpm.app.models.repository import RpmRepository
+
+log = getLogger(__name__)
+
+
+def prune_repo_nevras(repo_pk, keep_days, dry_run):
```

Comment:
`prune_repo_packages`

Reply:
Should we make both `n-to-keep` and the time threshold configurable?

Reply:
I'd probably start with simplest and always n1 to keep, we could make it configurable later if we get requests for it

Reply:
We already have `retain_package_versions` if you want "how many nevras to keep", and  `retain_repo_versions` to prune old repositoryversions. This is "get rid of nevras older than X" regardless of those settings. Do we *really* want to make this even *more* complicated?  Even if someone somewhere thinks it would be nifty, I think we're already at the edge of "too many choices".

Reply:
I am all in for a KISS ;)